### ISSUE TITLE
Need documentation (like a schema) on resultJsonOutputFile

### ISSUE THREAD

**Issue by Carmageddon** [angular/protractor#4960]
So, using Protractor 5.4.0 has various JSON structure outputs in the result file.
For example, "assertions" array seems to include only one value "passed" when its true.
When its value is false, it has two more values outputted: "stackTrace" and "errorMsg"

We are writing a parser to handle the JSON output file, and to be sure we are handling every possibility, we really need a schema or something that tells us definitely what are all possible structures we could encounter.
For example: Can "assertions" array include more than one object?

I have not found any documentation regarding this.

Thanks!

**Comment by GitSage** [comment]
I agree, there ought to be documentation, and I can't seem to find it. 

Digging through the source, I did find this:

```
export interface RunResults {
  taskId: number;
  specs: Array<string>;
  capabilities: any;
  failedCount: number;
  exitCode: number;
  specResults: Array<any>;
}
```

```
specResults: [{
  description: string,
  assertions: [{
    passed: boolean,
    errorMsg: string,
    stackTrace: string
  }],
  duration: integer
}]
```

**Comment by Carmageddon** [comment]
Thank you! Does it mean assertions array can contain multiple objects? Not sure about this syntax.. seems to be Go structure?

**Comment by GitSage** [comment]
This is Typescript. You can read about type annotations in Typescript [here](http://www.tutorialsteacher.com/typescript/type-annotation). `assertions` is an array of objects, each of which has three properties: `passed` (a boolean), `errorMsg` (a string), and `stackTrace` (another string). You can basically read it almost as if it were JSON.
